contractcourt/contract_resolvers.go

```diff
@@ -86,8 +86,10 @@ type ResolverConfig struct {
 
 	// Checkpoint allows a resolver to check point its state. This function
 	// should write the state of the resolver to persistent storage, and
-	// return a non-nil error upon success.
-	Checkpoint func(ContractResolver) error
+	// return a non-nil error upon success. It takes a resolver report,
+	// which contains information about the outcome and should be written
+	// to disk if non-nil.
+	Checkpoint func(ContractResolver, *channeldb.ResolverReport) error
```

Comment:
`PutResolverReport` in the arb config? We don't straight access to channeldb where we call `Checkpoint`, so `PutResolverReport` is added to that we can pass it down to checkpoint. It needs to be written in the same tx, which is why we pass down to `Checkpoint` rather than have a separate function.